src/ComunicaEngine.js

```diff
@@ -31,13 +36,21 @@ export default class ComunicaEngine {
     let bindings;
     const next = async () => {
       if (!bindings) {
-        // Determine the document to query from the subject if there is no source
-        const source = this._source ?
-          { type: 'rdfjsSource', value: this._source } :
-          { type: 'file', value: this.getDocument(await this._subject) };
+        let sources;
+        if (this._source) {
+          const value = await this._source;
+          if (Array.isArray(value))
```

Comment:
I would `if (!Array.isArray(value)) value = [value]` to have one code path for the handling

Reply:
…and then you can inline `handleSource`